server/src/weaverbird/backends/mongo_translator/steps/rank.py

```diff
@@ -0,0 +1,91 @@
+from typing import List
+
+from weaverbird.backends.mongo_translator.steps.types import MongoStep
+from weaverbird.pipeline.steps import RankStep
+
+
+def translate_rank(step: RankStep) -> List[MongoStep]:
+    vqb_var_order: MongoStep = {}
+
+    if step.method == 'dense':
+        vqb_var_order = {
+            '$cond': [
+                {'$ne': [f'$$this.{step.value_col}', '$$value.prevValue']},
```

Comment:
doesn't the double `$$` in a JS backtick string escape to a single `$` ?

Reply:
If am not wrong, yes it escape to a singe $, but in this case it was `$$this` not `$${this}` so if we take an example to see well the difference, it could look like this : 
![Screenshot from 2022-05-02 19-59-15](https://user-images.githubusercontent.com/22576758/166300011-11d428b5-319d-4738-9e84-6666c41b5734.png)